Source/Urho3D/AngelScript/ScriptAPIDump.cpp

```diff
@@ -386,11 +396,26 @@ void Script::DumpAPI(DumpMode mode, const String& sourceTree)
             }
             else if (mode == C_HEADER)
             {
-                ///\todo Find a cleaner way to do this instead of hardcoding
-                if (typeName == "Array")
-                    Log::WriteRaw("\ntemplate <class T> class " + typeName + "\n{\n");
-                else
-                    Log::WriteRaw("\nclass " + typeName + "\n{\n");
+                if (type->GetFlags() & asOBJ_TEMPLATE) {
+                    String str = "\ntemplate <";
+                    for (asUINT tt = 0, ttm = type->GetSubTypeCount(); tt < ttm; tt++) {
+                        asITypeInfo* pSubType = type->GetSubType(tt);
+                        str += String("class ") + pSubType->GetName() + (tt < ttm - 1 ? ", " : ">");
+                    }
+                    Log::WriteRaw(str);
+                }
+                Log::WriteRaw("\nclass " + typeName + "\n{\npublic:\n");
+            }
+            if (mode == C_HEADER)

```

Comment:
I don't really understand this code, but `else if (mode == C_HEADER) { ... }; if (mode == C_HEADER) { ... }` looks suspicious